docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Add-Providers-Support
+authors:
+  - "@Gal-Zaidman"
+creation-date: 2021-07-25
+last-updated: 2021-07-25
+---
+
+# Add Providers Support
+
+## Summary
+
+Currently, the assisted installer supports Bare-Metal installations only.
+Openshift can be installed on various providers such as:
+- Red Hat Virtualization.
+- OpenStack.
+- vSphere.
+- AWS.
+- GCP.
+- Azure.
+
+This enhancement will define a clear interface to extend the assisted installer and allow
+provider installations.
+
+## Motivation
+
+Assisted Installer can be a great tool to perform UPI (User Provisioned) installation on external providers,
+providing the user with clear UI to see its provisioned machines joining the cluster, and lowering installation
+resources requirements.
+
+Assisted Installer is also ideal for mixed env configurations.
+** The RHV provider supports installing a mixed env where the Control-Plain (Master) nodes are deployed on top of RHV
+and Compute (Worker) nodes are Bare-Metal nodes.
+
+### Goals
+
+- Create a Provider interface, and hooks in the assisted installer code which separates the assisted installer core code
```

Comment:
It would be an improvement to rephrase this as an outcome rather than a solution. The goal is to enable additional platforms in some way, so that a user can create a cluster with platform integration. What you're describing here is solution details that would be better in the Proposal section.

Reply:
I aimed for the [provider interface](https://github.com/openshift/assisted-service/pull/2272/files#diff-988c4e5d655472b2174837171f5d89978ca0f86ec8554329b386dccb9990ea83R84-R90) to do that.
I would expand it and update the enhancement as we will find more provider-specific parts, in the meantime, I relaied on the adding vSphere PR that was merged by the team

Reply:
OK I agree, moved to the proposal part and replaced with:
- Define a clear way to add a specific platform provider to the assisted installer.
- Add the oVirt platform provider to the assisted installer.
- Migrate all existing supported providers to implement the same unified interface.

Is that better?